### DISCUSSION TITLE
Instructions for DIY virtualization/remote setup?

### DISCUSSION THREAD

**Discussion by alshdavid** [containers/podman#15050] [Q&A]
With docker you can use the `DOCKER_HOST` variable and point that to a remote docker machine via ssh.

This allows us to manage our own Linux virtual machine (which allows us to consolidate multiple Linux-only services) or connect to an external networked (more powerful) machine.

Are there instructions for installing podman on a client and pointing it to an external podman host via ssh? Are there instructions on installing podman on Windows/MacOS without the VM so the container host can be manually managed?

**Comment by afbjorklund** [comment]
It is basically the same thing, but you use `CONTAINER_HOST` instead. You only need to install podman-remote, on the clients.

https://github.com/containers/podman/blob/main/docs/tutorials/remote_client.md

One small difference is that Podman requires adding the socket path, whereas Docker sets that up on the server (login environment)

https://docs.docker.com/engine/security/rootless/#expose-docker-api-socket-through-ssh

----

docker (rootless)
`systemctl --user enable --now docker.service`
`sudo loginctl enable-linger $(whoami)`
`DOCKER_HOST=ssh://<REMOTEUSER>@<REMOTEHOST> # unix:///run/user/1000/docker.sock`

podman (rootless)
`systemctl --user enable --now podman.socket`
`sudo loginctl enable-linger $USER`
`CONTAINER_HOST=ssh://<REMOTEUSER>@<REMOTEHOST>/run/user/1000/podman/podman.sock`

**Comment by rhatdan** [comment]
You can use podman system connection to manage this as well

**Reply by afbjorklund** [reply]
Similar to using docker context to manage those, same difference ?

The lima examples have some "translations" for both runtimes, for instance:

* https://github.com/lima-vm/lima/blob/master/examples/docker.yaml

`DOCKER_HOST`
`docker context create`
`docker context use`
`docker run`

* https://github.com/lima-vm/lima/blob/master/examples/podman.yaml

`CONTAINER_HOST`
`podman system connection add`
`podman system connection default`
`podman --remote run`

**Reply by rhatdan** [reply]
Yes we really need to get these commands alias.

podman context create -> podman system connection add
podman context use -> podman system connection default
